### PR TITLE
fix test_time failure in non-DST time zones

### PR DESCRIPTION
**Pull request by hujiajie**
Fixes: #5617

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -2083,7 +2083,16 @@ def test_tcgetattr(self):
   def test_time(self):
     src = open(path_from_root('tests', 'time', 'src.cpp'), 'r').read()
     expected = open(path_from_root('tests', 'time', 'output.txt'), 'r').read()
-    self.do_run(src, expected);
+    try:
+      for tz in ['EST+05EDT', 'UTC+0']:
```
Comment (kripken) [inline tests/test_core.py]: Can we keep the current execution without TZ set, that is, that this PR only adds 2 more executions? (can do it before the loop, if so)

File: src/library.js [hunk 2]
```diff
@@ -1913,7 +1913,7 @@ LibraryManager.library = {
     var winterOffset = start.getTimezoneOffset();
     var dstOffset = Math.min(winterOffset, summerOffset); // DST is in December in South
     if (dst < 0) {
-      {{{ makeSetValue('tmPtr', C_STRUCTS.tm.tm_isdst, 'Number(dstOffset == guessedOffset)', 'i32') }}};
+      {{{ makeSetValue('tmPtr', C_STRUCTS.tm.tm_isdst, 'Number(summerOffset != winterOffset && dstOffset == guessedOffset)', 'i32') }}};
```
Comment (kripken) [inline src/library.js]: Interesting, so the issue is that some regions don't have DST at all?

Please add a comment mentioning that.

File: tests/test_core.py (hunk 1 above)
Comment (hujiajie) [reply tests/test_core.py]: Done

File: src/library.js (hunk 2 above)
Comment (hujiajie) [reply src/library.js]: Done

**Comment by kripken** [comment]
Thanks!

**Pull request merged**